core/exportfuncs.cpp

```diff
@@ -0,0 +1,311 @@
+// SPDX-License-Identifier: GPL-2.0
+#include <QFileInfo>
```

Comment:
From a reviewer's perspective it is rather challenging to have one commit apparently add new export code that I need to go through line by line to understand, and then have almost the same code removed elsewhere two commits later.

What you should do is have one commit `move export functions into separate compilation unit` and allow the reviewer to easily see that this is the same code that just moved into a different file.

Reply:
I have updated the commit text, to state it is a 1-1 copy from divelogexportdialog. But I really believe (also from a reviewers perspective) to not make commit across directories, and in this case split the adding/removing (to keep the removing as simple as possible). If you do not agree I will squash the commits.

Reply:
That's interesting. I don't see "not make commit across directories" as fairly unimportant. To me it matters more to have both parts of the move in the same commit so I can see how this affects things.

But I do of course see your point that this way you can introduce the new code, then switch to using it, then remove the old code, which also has a lot of advantages.

No need to squash the commits, what you did isn't wrong.